src/rust/iced-x86/src/register_iterator.rs

```diff
@@ -0,0 +1,43 @@
+use core::intrinsics::transmute;
+use core::iter::Iterator;
+
+use super::iced_constants::IcedConstants;
+use super::Register;
+
+/// Iterator for `Register` enum values
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct RegisterIterator {
+	curr_val: u8,
+}
+
+impl RegisterIterator {
+	/// Creates a new iterator
+	#[must_use]
+	#[inline]
+	pub fn new() -> Self {
```

Comment:
This doesn't need to be public. Users should always call `Register::values()` to get the iterator.